### ISSUE TITLE
emit C references instead of pointers

### ISSUE THREAD

**Issue by Kamayuq** [eqrion/cbindgen#210]
shouldn't rust references like &T and &mut T generate const T& and T& respectively instead of pointers as rust references can never be nullptr?

**Comment by RReverser** [comment]
C doesn't have references. Did you mean C++?

**Comment by Kamayuq** [comment]
You are right it does not. Maybe we can have this as an option for the C++ backend? extern C functions with references work fine AFAIK.

**Comment by athre0z** [comment]
IIRC, the C++ standard doesn't provide any guarantees regarding reference representation, so receiving a reference from a FFI call would result in UB.

**Comment by emilio** [comment]
Itanium does define it: https://itanium-cxx-abi.github.io/cxx-abi/abi.html#reference-parameter

I'd be surprised if MSVC or other ABIs didn't do the same.
